### DISCUSSION TITLE
Is it possible to mix mermaid and code block highlighting in the same file?

### DISCUSSION THREAD

**Discussion by oinoom** [srid/emanote#281] [Q&A] [answered]
I have a simple page.headHtml:

```
  headHtml: |
    <link href="https://cdn.jsdelivr.net/npm/prismjs@1.28.0/themes/prism-tomorrow.css" rel="stylesheet" />
    <script src="https://cdn.jsdelivr.net/combine/npm/prismjs@1.28.0/prism.min.js,npm/prismjs@1.28.0/plugins/autoloader/prism-autoloader.min.js"></script>

    <script src="https://cdn.jsdelivr.net/npm/mermaid/dist/mermaid.min.js"></script>
    <script>
      mermaid.initialize({startOnLoad:false});
      mermaid.init(undefined,document.querySelectorAll(".mermaid"));
    </script>

    <script>
      window.MathJax = {
        startup: {
          ready: () => {
            MathJax.startup.defaultReady();
          }
        }
      };
    </script>
    <script async="" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
```

What I'm finding when including both the mermaid example from the tips page and some python code is that the prism script conflicts with the mermaid script.

![image](https://user-images.githubusercontent.com/8881397/166126050-19a3b0a8-d0d2-4944-b382-4a6ccbeb5f1c.png)

**Comment by srid** [comment]
There apparently is a way to tell Prism not to highlight certain blocks: https://github.com/PrismJS/prism/pull/2074#issuecomment-535930585

But I cannot figure out how to make Mermaid play correctly with it. If you are able, let me know - we can modify the [default snippet](https://github.com/srid/emanote/blob/da80bafa4d134f911242a1b9721b92b1742f2790/default/index.yaml#L78-L90) accordingly.

**Reply by srid** [reply]
Note: adding a custom class to markdown code blocks would be like this:

<img width="286" alt="image" src="https://user-images.githubusercontent.com/3998/166129890-a995f281-56f0-4dbf-a80c-4bc019bf78d9.png">

**Comment by oinoom** [comment, marked as answer]
I gave up -- it looks like [prism strips HTML](https://prismjs.com/faq.html#if-pre-existing-html-is-stripped-off-how-can-i-highlight), and I think that's the root of the problem. I decided to go with highlight.js instead and was able to get that to work:

```
js:
  # Syntax highlighting using highlight.js
  syntax: |
    <link rel="stylesheet"
      href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.5.1/styles/hybrid.min.css">
    <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.5.1/highlight.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.5.1/languages/haskell.min.js"></script>
    <script>hljs.highlightAll();</script>
```

![image](https://user-images.githubusercontent.com/8881397/166150718-356c3680-8142-4e2b-8e11-fc651c55a912.png)

**Reply by srid** [reply]
Does highlight.js have autoloading of language JS (like prism does), or do you have to manually specifying them?

**Reply by oinoom** [reply]
I believe they autoload a limited subset which you can see on their [download page](https://highlightjs.org/download/), and for languages like Haskell it needs to be manually specified.
